aws_lambda_script/app/resources.py

```diff
@@ -0,0 +1,619 @@
+                array_data[category] = [item.lower() for item in array_data[category]]
+                for item in items:
+                    item = item.lower()
+                    if item not in array_data[category]:
+                        array_data[category].append(item)
+
+        write_array_data(array_data)
+
+        return new_project, 201
+
+
+@ns.doc(
+    responses={
+        200: "Success",
+        401: "Authorization is required",
+        404: "Project not found.",
+    }
+)
+@ns.route("/projects/<string:project_name>")
+class ProjectDetail(Resource):
+    # Loop through all projects and return the one that matches the name and the user email in the first user item in the user list
+    @ns.marshal_list_with(get_project_model())
+    def get(self, project_name):
+        owner_email = get_user_email(parser.parse_args())
+
+        # Sanitize project_name by replacing '%20' with spaces
+        project_name = project_name.replace("%20", " ")
+
+        data = read_data()
+        project = next(
+            (
+                proj
+                for proj in data["projects"]
+                if proj["details"][0]["name"] == project_name
+                and any(user["email"] == owner_email for user in proj["user"])
+            ),
+            None,
+        )
+        if not project:
+            abort(404, description="Project not found")
+        return project, 200
+
+    # Edit a project by taking the whole schema and replacing the existing project with the same name and owner
+    @ns.marshal_list_with(get_project_model())
+    @ns.doc(
+        responses={
+            200: "Updated project",
+            401: "Authorization is required",
+            404: "Project not found",
+            406: "Missing JSON data",
+        }
+    )
+    def put(self, project_name):
+        owner_email = get_user_email(parser.parse_args())
+        updated_project = ns.payload
+
+        if (
+            "user" not in updated_project
+            or "details" not in updated_project
+            or "email" not in updated_project["user"][0]
+            or "name" not in updated_project["details"]
+        ):
+            abort(406, description="Missing JSON data")
+
+        # Ensure the email is set to owner_email
+        updated_project["user"][0]["email"] = owner_email
+
+        data = read_data()
+        project = next(
+            (
+                proj
+                for proj in data["projects"]
+                if proj["details"]["name"] == project_name
+                and any(user["email"] == owner_email for user in proj["user"])
+            ),
+            None,
+        )
+
+        if not project:
+            abort(404, description="Project not found")
+
+        # Update the project details
+        project.update(updated_project)
+        write_data(data)
+
+        return project, 200
+
+
+# Read the client keys from S3 bucket for Cognito
+cognito_settings = read_client_keys()
```

Comment:
As per the UI comment I think we want to be reading keys from secret manager

Reply:
Done 👍